### PR TITLE
Add test case for s390 boot menu

### PR DESCRIPTION
**Pull request by smitterl**
1. seabios not supported on arch, filter out
2. add test case display of qemu boot menu
3. add test case booting non default entry

Signed-off-by: Sebastian Mitterle <smitterl@redhat.com>

### CODE REVIEW

**Comment by smitterl** [comment]
```bash
# avocado run --vt-type libvirt --vt-machine-type s390-virtio --vt-connect-uri qemu:///system boot_integration.by_qemu_on_s390.boot_dev.boot_only
JOB ID     : 6194893deed4e8b46e2b5a82f514499b030f4271
JOB LOG    : /root/avocado/job-results/job-2020-07-16T14.57-6194893/job.log
 (1/1) type_specific.io-github-autotest-libvirt.boot_integration.by_qemu_on_s390.boot_dev.boot_only: PASS (93.06 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 98.07 s
```

**Comment by smitterl** [comment]
```bash
# avocado run --vt-type libvirt --vt-connect-uri qemu:///system boot_integration
JOB ID     : 27c15c844bf647c16d57265e95b2510804a3cfe2
JOB LOG    : /root/avocado/job-results/job-2020-07-16T21.06-27c15c8/job.log
 (1/2) type_specific.io-github-autotest-libvirt.boot_integration.by_seabios.boot_dev.define_start_destroy_save_restore_undefine: PASS (67.45 s)
 (2/2) type_specific.io-github-autotest-libvirt.boot_integration.by_seabios.boot_order.define_start_destroy_save_restore_undefine: PASS (65.70 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 134.59 s
```

File: libvirt/tests/src/bios/boot_integration.py [hunk 1]
```diff
@@ -56,13 +61,34 @@ def prepare_boot_xml(vmxml, params):
...
+def console_check(vm, pattern, debug_log=False):
+    """
+    Return function for use with wait_for.
+
+    :param vm: vm to check serial console for
+    :param pattern: line or pattern to match
+    :param debug_log: if to log all output for debugging
+    :return: function returning true if console output matches pattern
+    """
+    def matches():
```
Comment (chunfuwen) [inline libvirt/tests/src/bios/boot_integration.py:73]: change name matches() to _matches() ?

**Comment by smitterl** [comment]
```bash
avocado run --vt-type libvirt --vt-machine-type s390-virtio --vt-connect-uri qemu:///system boot_integration
JOB ID     : 850697bd8bac1e946d8ce441a73871ebffd425a9
JOB LOG    : /root/avocado/job-results/job-2020-07-17T03.17-850697b/job.log
 (1/1) type_specific.io-github-autotest-libvirt.boot_integration.by_qemu_on_s390.boot_dev.boot_only: PASS (116.11 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 118.07 s
```

**Comment by smitterl** [comment]
```bash
# avocado run --vt-type libvirt --vt-machine-type s390-virtio --vt-connect-uri qemu:///system boot_integration
/root/.libvirt-ci-venv-ci-runtest-LaAQRT/lib/python3.6/site-packages/avocado_framework-80.0-py3.6.egg/avocado/plugins/run.py:292: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast and keep-tmp. 
  FutureWarning)
JOB ID     : bbbb3d95a26888556f6f3ae97dbbd36e893239f7
JOB LOG    : /root/avocado/job-results/job-2020-07-17T05.49-bbbb3d9/job.log
 (1/2) type_specific.io-github-autotest-libvirt.boot_integration.by_qemu_on_s390.boot_dev.check_menu: PASS (73.91 s)
 (2/2) type_specific.io-github-autotest-libvirt.boot_integration.by_qemu_on_s390.boot_dev.boot_non_default: PASS (137.21 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 213.09 s
```

**Comment by smitterl** [comment]
@chunfuwen I modified based on your suggestion. I also added another test case, I hope you don't mind reviewing again. Thank you.

File: libvirt/tests/src/bios/boot_integration.py [hunk 2]
```diff
@@ -114,38 +142,63 @@ def run(test, params, env):
...
+            if boot_type == "ovmf":
+                if os.path.exists(nvram_file):
+                    test.fail("nvram file still exists after vm undefine")
     finally:
         logging.debug("Start to cleanup")
         if vm.is_alive:
             vm.destroy()
         logging.debug("Restore the VM XML")
-        vmxml_backup.define()
+        if not boot_type == "s390_qemu":
```
Comment (chunfuwen) [inline libvirt/tests/src/bios/boot_integration.py:201]: vmxml_backup.sync() is general purpose method , so no need  diff boot_type.
This is minor suggestion, anyway, I approve it.

Comment (smitterl) [reply libvirt/tests/src/bios/boot_integration.py:201]: @chunfuwen Do you mean we could just use `vmxml_backup.sync()` for all cases?

Comment (chunfuwen) [reply libvirt/tests/src/bios/boot_integration.py:201]: yes

Comment (smitterl) [reply libvirt/tests/src/bios/boot_integration.py]: @chunfuwen Got it. Thanks :) Updated.

**Comment by smitterl** [comment]
@dzhengfy Can you help merge this? Thank you.

**Pull request merged**